vernac/vernacentries.ml

```diff
@@ -1829,7 +1829,10 @@ let check_may_eval env sigma redexp rc =
   let uctx = Evd.universe_context_set sigma in
   let env = Environ.push_context_set uctx (Evarutil.nf_env_evar sigma env) in
   let { Environ.uj_val=c; uj_type=ty; } =
-    if Evarutil.has_undefined_evars sigma c then
+    if Evarutil.has_undefined_evars sigma c
+    || List.exists (Context.Named.Declaration.exists (Evarutil.has_undefined_evars sigma))
```

Comment:
You should use Evarutil.is_ground_env instead.

Reply:
That looks wrong, it doesn't check that the types are evar free
~~~ocaml
let is_ground_env evd env =
  let is_ground_rel_decl = function
    | RelDecl.LocalDef (_,b,_) -> is_ground_term evd (EConstr.of_constr b)
    | _ -> true in
  let is_ground_named_decl = function
    | NamedDecl.LocalDef (_,b,_) -> is_ground_term evd (EConstr.of_constr b)
    | _ -> true in
  List.for_all is_ground_rel_decl (rel_context env) &&
  List.for_all is_ground_named_decl (named_context env)
~~~

Reply:
Hmm, right. Now I'm a bit surprised that this function is written that way.